main.ml

```diff
@@ -329,7 +345,23 @@ let choose_datapath ?(persistent = true) response =
   | (scheme, u) :: us -> return (Ok (scheme, u, "0"))
 
 (* Process a message *)
-let process root_dir name x =
+let process root_dir name =
+  let version = ref None in
```

Comment:
This works because we invoke this thread for each plugin separately, so we'll have a different `version` reference for each plugin?

Reply:
Yes it gets partially applied with just a `root_dir` and a `name`: `Message_switch_async.Protocol_async.Server.listen ~process:(process root_dir name) ~switch:switch_path ~queue:(Filename.basename name) ()`.
So each plugin will get its own version.

Reply:
Should we mention this in a comment because I found it a bit tricky to understand from a quick read?

Reply:
Although now that I think about it it's obvious that we'll always have a unique ref whenever we use this function.